HGustavs/LenaSYS
# G2 2024 v4 issue#14553 Created processDuggaFile.php

Created _processDuggaFile.php_ solves issue [14553](https://github.com/HGustavs/LenaSYS/issues/14553).

## OVERVIEW
This module/component should not be considered a microservice; it functions as a supporting tool for a potential microservice  (if there is not one already) that would display a dugga. Its primary role is to facilitate the retrieval and processing of dugga files.

## How to test:
1. Open showDuggaservice.php located in DuggaSys.
2. Add `include_once "../DuggaSys/microservices/showDuggaService/processDuggaFile.php";`
3. Comment out the function `processDuggaFiles();` within the codeblock located approximately between lines 129 and 247
4. Locate the references to the function, add following parameters `$pdo, $courseid,$coursevers,$duggaid,$duggainfo, $moment` and declare it to a variable.
_Example:_ `$submittedFiles = processDuggaFiles($pdo, $courseid,$coursevers,$duggaid,$duggainfo, $moment);`
5. Check if the `$submittedFiles` is not empty and declare the $files to the new variable above declaring the array near the bottom of the code. 
6. Go to a Dugga where it is possible to upload a file.
7. See if the file is there and check returned data inside `showDuggaservice.php` in devTools
